src/mixins/segmentation/freehandSegmentationMixin.js

```diff
@@ -0,0 +1,189 @@
+import external from '../../externalModules.js';
+import store from '../../store';
+import { getLogger } from '../../util/logger.js';
+import { draw, drawJoinedLines, getNewContext } from '../../drawing';
+
+const logger = getLogger('tools:ScissorsTool');
+
+const { getters, setters } = store.modules.segmentation;
+
+/**
+ * Render hook: draws the FreehandScissors's outline
+ *
+ * @param {Object} evt Cornerstone.event#cornerstoneimagerendered > cornerstoneimagerendered event
+ * @memberof Tools.ScissorsTool
+ * @returns {void}
+ */
+function renderToolData(evt) {
+  const eventData = evt.detail;
+  const { element } = eventData;
+  const color = getters.brushColor(element, true);
+  const context = getNewContext(eventData.canvasContext.canvas);
+  const handles = this.handles;
+
+  draw(context, context => {
+    if (handles.points.length > 1) {
+      for (let j = 0; j < handles.points.length; j++) {
+        const lines = [...handles.points[j].lines];
+        const points = handles.points;
+
+        if (j === points.length - 1) {
+          // If it's still being actively drawn, keep the last line to
+          // The mouse location
+          lines.push(this.handles.points[0]);
+        }
+        drawJoinedLines(context, element, this.handles.points[j], lines, {
+          color,
+        });
+      }
+    }
+  });
+}
+
+/**
+ * Sets the start handle point and claims the eventDispatcher event
+ *
+ * @private
+ * @param {*} evt // mousedown, touchstart, click
+ * @returns {void|null}
+ */
+function _startOutliningRegion(evt) {
+  const element = evt.detail.element;
+  const image = evt.detail.currentPoints.image;
+
+  if (!this.handles.points.length) {
+    logger.warn('Something went wrong, empty handles detected.');
+
+    return null;
+  }
+
+  this.handles.points.push({
+    x: image.x,
+    y: image.y,
+    lines: [],
+  });
+
+  this.currentHandle += 1;
+
+  external.cornerstone.updateImage(element);
+}
+
+/**
+ * This function will update the handles and updateImage to force re-draw
+ *
+ * @private
+ * @method _setHandlesAndUpdate
+ * @param {(CornerstoneTools.event#TOUCH_DRAG|CornerstoneTools.event#MOUSE_DRAG|CornerstoneTools.event#MOUSE_MOVE)} evt  Interaction event emitted by an enabledElement
+ * @returns {void}
+ */
+function _setHandlesAndUpdate(evt) {
+  const eventData = evt.detail;
+  const element = evt.detail.element;
+
+  this._addPoint(eventData);
+  external.cornerstone.updateImage(element);
+}
+
+/**
+ * Event handler for MOUSE_UP/TOUCH_END during handle drag event loop.
+ *
+ * @private
+ * @method _applyStrategy
+ * @param {(CornerstoneTools.event#MOUSE_UP|CornerstoneTools.event#TOUCH_END)} evt Interaction event emitted by an enabledElement
+ * @returns {void}
+ */
+function _applyStrategy(evt) {
```

Comment:
I'm not convinced this is the best way to go about this. Flagging this for a conversation -- we should unpack what we're trying to accomplish here and make sure this is the right way.

Reply:
_FLAGGING_

Reply:
(I already changed this as I hate it too)